lib/spack/docs/known_issues.rst

```diff
@@ -88,3 +88,19 @@ software outside of Spack. Unfortunately, this command no longer works.
 See https://github.com/spack/spack/issues/2597 and
 https://github.com/spack/spack/issues/2662 for details. This is expected
 to be fixed by https://github.com/spack/spack/pull/2664.
+
+--------------------------------
+intel-mkl does not work on macOS
+--------------------------------
+
+**Status:** Only a workaround available due to the way Intel provides
+the library
+
+Installing intel-mkl under macOS requires a different download. This
+is not compatible with spack. However, there exists a workaround which
+is as follows:
+
+1. Download the intel-mkl version for macOS from Intel's homepage.
+2. Install the downloaded version manually.
```

Comment:
There are a few additional wrinkles with a Mac installation.  You may want add the following caveat to the doc here:

-----
MKL _can not_ be installed as regular user – administrator privileges are required.

Moreover, the user running the MKL GUI installer is assumed to either (a) be an administrator (able to `sudo`), or (b) has set a password for `root` on the local machine (which is not a given on Mac).

If neither is the case, choose one of the following:
* Log in as administrator, then run the m_mkl*.app normally,
**or**
* Run the following shell command directly:
```
sudo /Volumes/m_mkl_2018.1.126/m_mkl_2018.1.126.app/Contents/MacOS/install.sh
```
(Adapt the version numbers as appropriate.)

-----

Here's the evidence: Starting the GUI installer as regular user brings up the following:
![screen shot 2018-03-17 at 14 56 02](https://user-images.githubusercontent.com/31289496/37559408-a68510a8-29f3-11e8-8d56-72cb1cf0b746.png)
Observe how there is no means to supply an admin _user name_ as would normally be done if the system-native elevation dialog were used – sigh. Of lesser importance is the following, though also a bit of an eye-roller:
```
$ ./install.sh --cli-mode

Graphical user interface must be enabled to run this program.

Quitting!
```

Reply:
@f-sim could you please add this comment?